sheets/3/exercise-2.tex

```diff
@@ -1,11 +1,107 @@
-    \excercise{Schleifen}
-            \subexcercise Schreibe deine \texttt{turnLeft()} so um, dass sie eine for-Schleife verwendet.
-
-            \subexcercise Neo muss so schnell wie möglich die nächstgelegene Telefonzelle erreichen, um die Matrix zu verlassen. Leider weiß er nicht genau, wie weit diese entfernt ist. Schreibe hierzu eine Operation, die Neo so lange geradeaus laufen lässt, bis die Operation \texttt{Neo.isOnPhoneBooth()} den Wert \texttt{true} zurück gibt. Verwende in deiner Lösung eine while-Schleife.
-
-            \subexcercise(\textbf{Bonus}) Schreibe die Operation aus \textit{(b)} so um, dass sie eine do-while-Schleife verwendet.
-
-            \subexcercise Dieses Mal haben die Agenten alle Telefonzellen in Neos Nähe zerstört. Welche Auswirkung hat das auf deinen Code aus \textit{(b)}? Nutze die Operation \texttt{Neo.canMove()}, um den Fehler zu beheben.
-
-            \textbf{Für diese Aufgabe} müsst ihr \texttt{this.boothsDestroyed} auf \texttt{true} setzen.
-
+\excercise{Schleifen}
+	\subsection*{a)}
+		Passe die Main Methode an die Aufgabe an.
+	\subsection*{b)}
+		Wir möchten wissen wie viele Münzen auf dem Feld sind auf dem sich Neo befindet. Dafür heben wir so lange immer eine Münze auf bis sich keine Münzen mehr auf dem Feld befinden. Wenn die Aufgehobenen Münzen zählen können wir somit 				herausfinden wie viele Münzen auf dem Feld sind.
+		\begin{lstlisting}
+/**
+*
+* @param neo the neo whose Position is to be used
+* @return the number of coins on the Field that neo stands on
+*/
+public int countCoins(Neo neo){
+	int coins = 0;				//Zählvariable auf 0 setzen
+	while(neo.canCollectCoin()){//wenn Neo eine Münze aufheben kann
+		neo.collectCoin();		//hebe eine Münze auf
+		coins++;				//und erhöhe die Zählvariable um 1
+	}
+	for(int i = 0; i < coins; i++){ 	//lässt alle aufgesammelten
+		neo.dropCoin();					//Münzen wieder fallen.
+	}		//denn eine Abfrage sollte nicht das Spielfeld verändern
+	return coins;				//gebe als Ergebniss die 
+}								//Zählvariable zurück
+\end{lstlisting}
+
+\begin{Infobox}[If-Conditions 2]
+If-Conditions kennt ihr ja schon. Diese können auch dazu verwendet Zahlen (z.B Integer) miteinander zu vergleichen.
+\begin{lstlisting}
+int x = 2;
+if ( x < 3 ) {
+	System.out.println("x ist kleiner als 3!");
+}
+\end{lstlisting}
+Das kann man zum Beispiel dazu verwenden um zu schauen wie viele Münzen Neo hat. Aber Achtung! Java Verwendet die folgenden vergleichssymbole:
+\begin{center}
+        \begin{tabular}{ l | c | c | l }
+                Text & Math. Zeichen & Java & Bemerkung\\
+            \hline
+                  gr\"oßer als & $>$ & $>$ & \\
+                  kleiner als & $<$ & $<$ & \\
+                  gleich & $=$ & $==$ & einfaches $=$ ist in Java eine Zuweisung\\

```

Comment:
```suggestion
                  gleich & $=$ & $==$ & mit dem doppelten $==$ sollte man nur Zahlen und keine Objekte vergleichen! Ein einfaches $=$ ist in Java eine Zuweisung!\\

```

Vielleicht sollte man die Anmerkung hier in eine Rote Infobox auslagern "Vergleiche und Objekte" und da die `.equals` Methode erklären.

Reply:
Würd ich erst machen wenn die `.equals` auch brauchen.

Reply:
Die Info ist so wichtig, dass ich denke es ist gerechtfertigt wenn wir das vorzeitig erklären. Sobald sie == kennen werden sie damit auch Objekte vergleichen wollen. (Aber man könnte .equals erstmal weglassen und nur erklären, dass == nur für Zahlenwerte tut.) Auslagern in eine Infobox würde ich es trotzdem.